src/openrct2-ui/windows/LandRights.cpp

```diff
@@ -45,409 +46,361 @@ static rct_widget window_land_rights_widgets[] = {
- *
- *  rct2: 0x006681FB
- */
-static void WindowLandRightsTooldrag(rct_window* w, rct_widgetindex widgetIndex, const ScreenCoordsXY& screenCoords)
-{
-    if (_landRightsMode == LAND_RIGHTS_MODE_BUY_LAND)
+    void OnToolDown(rct_widgetindex widgetIndex, const ScreenCoordsXY& screenCoords) override
     {
-        if (screenCoords.x != LOCATION_NULL)
+        if (_landRightsMode == LAND_RIGHTS_MODE_BUY_LAND)
         {
-            auto landBuyRightsAction = LandBuyRightsAction(
-                { gMapSelectPositionA.x, gMapSelectPositionA.y, gMapSelectPositionB.x, gMapSelectPositionB.y },
-                LandBuyRightSetting::BuyLand);
-            GameActions::Execute(&landBuyRightsAction);
+            if (screenCoords.x != LOCATION_NULL)
+            {
+                auto landBuyRightsAction = LandBuyRightsAction(
+                    { gMapSelectPositionA.x, gMapSelectPositionA.y, gMapSelectPositionB.x, gMapSelectPositionB.y },
+                    LandBuyRightSetting::BuyLand);
+                GameActions::Execute(&landBuyRightsAction);
+            }
+        }
+        else
+        {
+            if (screenCoords.x != LOCATION_NULL)
+            {
+                auto landBuyRightsAction = LandBuyRightsAction(
+                    { gMapSelectPositionA.x, gMapSelectPositionA.y, gMapSelectPositionB.x, gMapSelectPositionB.y },
+                    LandBuyRightSetting::BuyConstructionRights);
+                GameActions::Execute(&landBuyRightsAction);
+            }
         }
     }
-    else
+
+    void OnToolDrag(rct_widgetindex widgetIndex, const ScreenCoordsXY& screenCoords) override
     {
-        if (screenCoords.x != LOCATION_NULL)
+        if (_landRightsMode == LAND_RIGHTS_MODE_BUY_LAND)
         {
-            auto landBuyRightsAction = LandBuyRightsAction(
-                { gMapSelectPositionA.x, gMapSelectPositionA.y, gMapSelectPositionB.x, gMapSelectPositionB.y },
-                LandBuyRightSetting::BuyConstructionRights);
-            GameActions::Execute(&landBuyRightsAction);
+            if (screenCoords.x != LOCATION_NULL)
+            {
+                auto landBuyRightsAction = LandBuyRightsAction(
+                    { gMapSelectPositionA.x, gMapSelectPositionA.y, gMapSelectPositionB.x, gMapSelectPositionB.y },
+                    LandBuyRightSetting::BuyLand);
+                GameActions::Execute(&landBuyRightsAction);
+            }
+        }
+        else
+        {
+            if (screenCoords.x != LOCATION_NULL)
+            {
+                auto landBuyRightsAction = LandBuyRightsAction(
+                    { gMapSelectPositionA.x, gMapSelectPositionA.y, gMapSelectPositionB.x, gMapSelectPositionB.y },
+                    LandBuyRightSetting::BuyConstructionRights);
+                GameActions::Execute(&landBuyRightsAction);
+            }
         }
     }
-}
 
-static bool LandRightsToolIsActive()
+private:
+    uint8_t _landRightsMode;
+    money32 _landRightsCost;
+    void InputSize()
```

Comment:
add a newline to split the memeber vars from the functions.

Reply:
Fixed.